external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpoutRetryExponentialBackoff.java

```diff
@@ -60,17 +61,19 @@ public int compare(RetrySchedule entry1, RetrySchedule entry2) {
             if(result == 0) {
                 //TreeSet uses compareTo instead of equals() for the Set contract
                 //Ensure that we can save two retry schedules with the same timestamp
-                result = entry1.hashCode() - entry2.hashCode();
+                result = entry1.instanceNumber - entry2.instanceNumber;
```

Comment:
Will revert to using hashCode(). I'd like to at least keep that check, since it would be annoying in tests using simulated time if we had to add 1 nanosecond between every call to spout.fail().